File: 1DAM/1DAM-Programacion/Unidad-6/Ejercicios/main-06-02-drodriguez/src/es/drodriguez/com/Repositories/ICRUDRepository.java
package es.drodriguez.com.Repositories;

import es.drodriguez.com.Models.Ficha;

import java.util.List;

public interface ICRUDRepository<T, ID> {
    List<T> findAll();

    T insert(T entity);

    T findById(ID id);

    Ficha delete(Ficha ficha);
}

File: 2DAM/AccesoADatos/Unidad-3/RA2_RA3_RA4_BBDDRelacionales/src/main/kotlin/repositories/ICrudRepository.kt
package repositories

import models.VehiculoDto

interface ICrudRepository<ID, T> {
    fun findAll(): List<VehiculoDto>
    fun exixstsById(id: ID): Boolean
    fun create(vehiculo: VehiculoDto): Int
    fun findById(id: ID): T?
    fun dropById(id: ID): Boolean
    fun findByUuid(uuid: String): T?
    fun updateByUuid(vehiculo: VehiculoDto): Boolean
}